### PR TITLE
Cleanup deprecations to develop next major version (v2)

### PR DESCRIPTION
**Pull request by sjoerdtalsma**
Main tasks to create new development baseline for next major release (v2).

- [x] Java version bumped to Java8 mimimum
- [x] `ContextManager.getActiveContext()` replaced by `getActiveContextValue()`.
- [x]  `ContextManager.clear()` ~~will require an implementation.~~ is part of ContextManager interface.
- [x]  All `@Deprecated(forRemoval=true)` is to be removed.
- [x] Document `@since 2.0.0` for new functionality

### CODE REVIEW

**Comment by sjoerdtalsma** [comment]
@Marcono1234 I hate to ask, but since you contributed before, could you have a look at this pull request I have prepared as new baseline for a greatly simplified, quicker 2.0 version?

By no means you have to do a full, detailed review (although welcome).
A quick opinion on the API changes would already be greatly appreciated! 🙏

**Review by Marcono1234 — COMMENTED** [review]
Sorry for the delay. I am currently not using this library, so I don't know how the changes and the new API feels for users. However, to me the changes look good and reasonable because they really simplify multiple parts.

I did not have an in-depth look at the changes, but hopefully the review comments are useful nonetheless.

File: timers/context-timer-micrometer/pom.xml [hunk 1]
```diff
@@ -23,13 +23,15 @@
...
 
     <!-- Artifact identification -->
-    <artifactId>context-propagation-micrometer</artifactId>
-    <name>Context propagation (micrometer)</name>
+    <groupId>nl.talsmasoftware.context.timers</groupId>
+    <artifactId>context-timer-micrometer</artifactId>
+    <name>Context timer (micrometer)</name>
     <packaging>jar</packaging>
 
     <properties>
```
[the remark this answers is not in the document]
Comment (Marcono1234) [reply timers/context-timer-micrometer/pom.xml:37]: Affects the new OpenTelemetry module as well, see https://github.com/talsma-ict/context-propagation/pull/516#discussion_r1843023894

**Comment by sjoerdtalsma** [comment]
> Sorry for the delay. I am currently not using this library, so I don't know how the changes and the new API feels for users. However, to me the changes look good and reasonable because they really simplify multiple parts.
> 
> I did not have an in-depth look at the changes, but hopefully the review comments are useful nonetheless.

@Marcono1234 thank you so much for the review and the many issues you caught!

I am contemplating on creating actual modules instead of the automatic modulenames currently used.
But those will definitely require tests with both a java8 and java 9+ test project, so still quite a bit of work, also because the serviceloader mechanism works differntly then.

File: context-propagation-core/src/main/java/nl/talsmasoftware/context/core/delegation/DelegatingExecutorService.java [hunk 2]
```diff
@@ -79,7 +80,7 @@ protected <T> Collection<? extends Callable<T>> wrapTasks(Collection<? extends C
             final List<Callable<T>> copy = new ArrayList<Callable<T>>(tasks.size());
             for (Callable<T> task : tasks) {
                 final Callable<T> wrapped = wrap(task);
-                modification |= (task == wrapped || (task != null && task.equals(wrapped))); // TODO Objects.equals
+                modification |= (task == wrapped || Objects.equals(task, wrapped));
```
[the remark this answers is not in the document]
Comment (sjoerdtalsma) [reply context-propagation-core/src/main/java/nl/talsmasoftware/context/core/delegation/DelegatingExecutorService.java:83]: Yes you're right, as the List has already been created at that point, there is no real perfomance benefit to not returning it.

**Comment by Marcono1234** [comment]
Just in case you aren't aware of it, the [ModiTect Maven Plugin](https://github.com/moditect/moditect/?tab=readme-ov-file#adding-a-module-descriptor-to-the-project-jar) can be used to add a module descriptor to the JAR. And if you specify `<jvmVersion>9</jvmVersion>`, then it will be a Multi-Release JAR, and the `module-info.class` should (in most cases) not cause issues for Java 8 users of your library.

But I am not sure if it is worth it adding a module descriptor instead of using automatic module names at the moment.

**Pull request merged**